### ISSUE TITLE
Menhir mode promote will generate .mock file in source directory

### ISSUE THREAD

**Issue by hackwaly** [ocaml/dune#1781]
Which it's unwanted.

```
(menhir
 (merge_into parser)
 (modules tokens parser)
 (flags --external-tokens Tokens)
 (mode promote))
```

```
src
├── dune
├── parser.ml
├── parser.mli
├── parser.mly
├── parser__mock.ml.mock
├── tokens.ml
├── tokens.mli
└── tokens.mly
```

**Comment by diml** [comment]
Fixed in #1783

**Comment by hackwaly** [comment]
Thank you!
